src/shim/error.rs

```diff
@@ -22,11 +25,63 @@ use serde::{Deserialize, Serialize};
 /// assert_eq!(shim_from_v3, fvm3_success.into());
 /// ```
 #[derive(PartialEq, Eq, Debug, Clone, Copy, Serialize, Deserialize, JsonSchema)]
-pub struct ExitCode(#[schemars(with = "u32")] ExitCodeV3);
+pub struct ExitCode(#[schemars(with = "u32")] ExitCodeV4);
+
+impl PartialOrd for ExitCode {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.value().cmp(&other.value()))
+    }
+}
+
+impl fmt::Display for ExitCode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let name = match self.0 {
+            ExitCode_latest::SYS_SENDER_INVALID => Some("SysErrSenderInvalid"),
+            ExitCode_latest::SYS_SENDER_STATE_INVALID => Some("SysErrSenderStateInvalid"),
+            ExitCode_latest::SYS_ILLEGAL_INSTRUCTION => Some("SysErrIllegalInstruction"),
+            ExitCode_latest::SYS_INVALID_RECEIVER => Some("SysErrInvalidReceiver"),
+            ExitCode_latest::SYS_INSUFFICIENT_FUNDS => Some("SysErrInsufficientFunds"),
+            ExitCode_latest::SYS_OUT_OF_GAS => Some("SysErrOutOfGas"),
+            ExitCode_latest::SYS_ILLEGAL_EXIT_CODE => Some("SysErrIllegalExitCode"),
+            ExitCode_latest::SYS_ASSERTION_FAILED => Some("SysFatal"),
+            ExitCode_latest::SYS_MISSING_RETURN => Some("SysErrMissingReturn"),
+
+            ExitCode_latest::USR_ILLEGAL_ARGUMENT => Some("ErrIllegalArgument"),
+            ExitCode_latest::USR_NOT_FOUND => Some("ErrNotFound"),
+            ExitCode_latest::USR_FORBIDDEN => Some("ErrForbidden"),
+            ExitCode_latest::USR_INSUFFICIENT_FUNDS => Some("ErrInsufficientFunds"),
+            ExitCode_latest::USR_ILLEGAL_STATE => Some("ErrIllegalState"),
+            ExitCode_latest::USR_SERIALIZATION => Some("ErrSerialization"),
+            ExitCode_latest::USR_UNHANDLED_MESSAGE => Some("ErrUnhandledMessage"),
+            ExitCode_latest::USR_UNSPECIFIED => Some("ErrUnspecified"),
+            ExitCode_latest::USR_ASSERTION_FAILED => Some("ErrAssertionFailed"),
+            ExitCode_latest::USR_READ_ONLY => Some("ErrReadOnly"),
+            ExitCode_latest::USR_NOT_PAYABLE => Some("ErrNotPayable"),
+
+            _ => None,
+        };
+        if let Some(name) = name {
+            write!(f, "{}({})", name, self.value())
+        } else {
+            match self.value() {
+                code if code > ExitCode_latest::SYS_MISSING_RETURN.value()
+                    && code < ExitCode_latest::FIRST_USER_EXIT_CODE =>
+                {
+                    write!(f, "SysErrReserved{}({})", code - 10, code)
```

Comment:
Why subtract 10?

Reply:
We want to align errors with Lotus display, So `code - 10` does that.

Maybe using `SYS_ASSERTION_FAILED` code instead? (it equals to 10).

Reply:
Do you know why Lotus subtracts 10? I'm okay with having it as well. Just please comment in the code the reason for this so the magic number is less magical.

Reply:
Sure, I will explain in a comment the rational behind this.

Reply:
I think there is a better, cleaner and more correct solution to match Lotus exit codes.

We just need to create a PR in ref-fvm to uncomment those dead lines:
https://github.com/filecoin-project/ref-fvm/blob/master/shared/src/error/mod.rs#L64
https://github.com/filecoin-project/ref-fvm/blob/master/shared/src/error/mod.rs#L74
https://github.com/filecoin-project/ref-fvm/blob/master/shared/src/error/mod.rs#L81-L84

And add those variant to our match expression in `forest/src/shim/error.rs`.

Reply:
Let's address that in a subsequent PR.